instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -176,15 +176,22 @@ def _before_request():
             return
         flask_request_environ = flask.request.environ
         span_name = get_default_span_name()
-        token = context.attach(
-            extract(flask_request_environ, getter=otel_wsgi.wsgi_getter)
-        )
+
+        token = ctx = None
+        span_kind = trace.SpanKind.INTERNAL
+
+        if trace.get_current_span() is trace.INVALID_SPAN:
+            ctx = extract(flask_request_environ, getter=otel_wsgi.wsgi_getter)
+            token = context.attach(ctx)
+            span_kind = trace.SpanKind.SERVER
 
```

Comment:
If span is found in current context, remote span is assume to be existing, so I believe you'll have to set the current context as the parent.

Reply:
Done.
I think it was passing earlier as `tracer.start_span()` makes current span as parent if context passed is `none`. 
@lzchen Should we set parent explicitly or let `start_span()` handle it?

Reply:
Oh yes you are correct. Either way is fine with me.